packages/react-tc-ui-kit/src/components/Input/Input.css

```diff
@@ -1,30 +1,108 @@
 .tc-input {
-  display: inline-block;
-  border: 1px solid rgba(43, 53, 83, 0.24);
-  border-radius: 6px;
-  padding: 0 16px;
+  position: relative;
   width: 100%;
-  height: 38px;
   box-sizing: border-box;
-  box-shadow: none;
+}
+
+.tc-input * {
+  box-sizing: border-box;
+}
+
+.tc-input__label {
+  display: block;
+  margin-bottom: 6px;
+  line-height: 1.2;
+}
+
+.tc-input__box {
+  position: relative;
+  width: 100%;
+}
+
+.tc-input__control {
+  display: block;
+  outline: none;
+  border: 1px solid #bac4ca;
+  border-radius: 10px;
+  padding: 0 16px;
+  width: 100%;
+  height: 42px;
   font-family: inherit;
+  font-weight: inherit;
+  font-size: inherit;
+  color: inherit;
+  background-color: #fff;
+}
+
+.tc-input__control:hover {
+  border-color: #81898e;
+}
+
+.tc-input__control:focus {
+  border-color: #2f3436;
+}
+
+.tc-input__control:disabled {
+  border-color: #bac4ca;
+  opacity: 0.5;
+}
+
+.tc-input__control--password {
+  padding-right: 46px;
+}
+
+.tc-input__view-toggler {
+  position: absolute;
+  right: 16px;
+  top: 11px;
+  outline: none;
+  border: none;
+  padding: 0;
+  width: 25px;
+  height: 25px;
   font-size: 16px;
   color: inherit;
   background-color: transparent;
+  cursor: pointer;
 }
 
-.tc-input:focus {
-  outline: none;
-  border-color: rgba(43, 53, 83, 0.5);
+.tc-input__view-toggler:hover,
+.tc-input__view-toggler:focus {
+  opacity: 0.6;
+}
+
+.tc-input__control:disabled ~ .tc-input__view-toggler {
+  pointer-events: none;
+}
+
+.tc-input--invalid .tc-input__box::after {
+  content: "";
+  position: absolute;
+  right: -6px;
+  top: 6px;
+  bottom: 6px;
+  width: 2px;
+  background-color: #e11961;
+}
+
+.tc-input__svg {
```

Comment:
place it right after view-toggler\
rename to `tc-input__view-toggler-icon`\
set `fill: currentColor`, do not hardcode it pls, we want it filled with current text color